instruqt/protecting-your-assets/05-license-entitlements/assignment.md

```diff
@@ -0,0 +1,202 @@
+      <td>
+        Since each boolean value represents whether an entitlement is true or
+        false, it's great for identifying capabilities that are turned on or
+        off in your application based in license terms. They can also be used
+        in a more structured way to represent a pricing model based in edition
+        by having specific boolean fields for each edition.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+### Another Use for License Fields
+
+One interesting topic that often comes up is what to do when you already have a
+licensing model in place. The Replicated Platform license is required for the
+platform to function, but your existing license is required for you code to
+work. This is another great use for Replicated License fields. You can use a
+text field to provide your existing license key and use the resulting field to
+provide the license to your code.
+
+Adding Your Own Entitlements
+============================
+
+Slackernews is tightly integrated to a customer's Slack team. It becomes more
+valuable with more Slack activity, and focuses on the links the team is
+sharing. Those three things suggest a few different measures that Slackernews
+might use to define it's license entitlements.
+
+* Members of the Slack team
+* Total volume of Slack messaging
+* Number of links the application is handling
+
+Defining a good business model for Slackernews is beyond the scope of this
+lab. Let's assume that members of the Slack team is the choice that Slackernews
+and it's stakeholders have decided on.
+
+### Adding the License Field
+
+You may still be logged into the Vendor Portal, if not log in again.
+
+Username: `[[ Instruqt-Var key="USERNAME" hostname="shell" ]]`<br/>
+Password: `[[ Instruqt-Var key="PASSWORD" hostname="shell" ]]`
+
+Pick "License Fields" from the menu on the left, then click either "+ Create
+license field" or "Create your first license field" to add a field.
+
+![Creating your first license field](../assets/create-a-license-field.png)
+
+Fill in the form with the following details and click "Create".
+
+Field: `member_count_max`
+Title: `Member Count`
+Type: `Integer`
+Default: `100`
+Required: checked
+
+This adds our Slack team member count entitlement to the license and makes it
+required on all licenses. Providing a default simplifies setup if the
+entitlement has a default value in your contracts. The default will also be
+used for your existing customers.
+
+### Setting the Entitlement Value
+
+We have three customers defined for Slackernews. Right now they all have the
+default value set for the new field. But Geeglo's contract is for 1000 members
+in it's team. Let's update their license.
+
+Click on "Customers" then "Geeglo" to bring up Geeglo. Bring up "Manage
+Customer" to edit their license details, including your new field. The custom
+fields are at the bottom, so you may need to scroll.
+
+![Updating Geeglo's license entitlements](../assets/updating-entitlements.png
```

Comment:
broken link, this shows up as is not as a link or image

Reply:
Interesting. The image is in the right place. I'm going to leave this one for the morning.

Reply:
There was a paren missing at the end of the image reference, so no actual image reference.